File: fortran/calc/calc.f90
PROGRAM calc
	IMPLICIT NONE
	DOUBLE PRECISION :: num1, num2, result
	INTEGER :: choice

	PRINT *, "CALC"
	PRINT *, "==== ==== ==== ====|"
	PRINT *, "1. Addition"
	PRINT *, "==== ==== ====    =|"
	PRINT *, "2. Subtraction"
	PRINT *, "==== ==== ====   = |"
	PRINT *, "3. Multiplication"
	PRINT *, "==== ==== ====   ==|"
	PRINT *, "4. Division"
	PRINT *, "==== ==== ====  =  |"
	PRINT *, "5. Square"
	PRINT *, "==== ==== ====  = =|"
	PRINT *, "6. Square Root"
	PRINT *, "==== ==== ====  == |"
	PRINT *, "7. Qube"
	PRINT *, "==== ==== ====  ===|"
	PRINT *, "Enter operation (1-7): "
	READ *, choice

	IF(choice == 1 .or. choice == 2 .or. choice == 3 .or. choice == 4)then
		PRINT *, "Enter first number: "
		READ *, num1
		PRINT *, "Enter second number: "
		READ *, num2
	ELSE
		PRINT *, "Enter number: "
		READ *, num1
	END IF

	SELECT CASE (choice)

	CASE (1)
		result = num1 + num2
		PRINT *, result
	CASE (2)
		result = num1 - num2
		PRINT *, result
	CASE (3)
		result = num1 * num2
		PRINT *, result
	CASE (4)
		IF(num2 /= 0)then
			result = num1 / num2
			PRINT *, result
		ELSE
			PRINT *, "Divide by zero error"
		END IF
	CASE (5)
		result = num1 * num1
		PRINT *, result
	CASE (6)
		result = SQRT(num1)
		PRINT *, result
	CASE (7)
		result = num1 * num1 * num1
		PRINT *, result
	CASE DEFAULT
		PRINT *, "Invalid operation"
	END SELECT
END PROGRAM calc

File: cpp/calc/calc.cpp
#include <iostream>
#include <cmath>

float result, num1, num2, only, pwr;
int func;

void get_function(){
	std::cout << "---- ---- ---- ----\n";
	std::cout << "calc\n";
	std::cout << "---- ---- ---- ----\n";
	std::cout << "\n";
	std::cout << " 1.Addition\n";
	std::cout << " 2.Subtraction\n";
	std::cout << " 3.Multiplication\n";
	std::cout << " 4.Division\n";
	std::cout << " 5.Exponents\n";
	std::cout << " 6.Square Root\n";
	std::cout << "\n";
	std::cout << "---- ---- ---- ----\n";
	std::cout << "Choose a function: ";
	std::cin >> func;
	std::cout << "\n";
	}

void get_both_inputs(){
	std::cout << "First number: 	";
	std::cin >> num1;
	std::cout << "Second number: 	";
	std::cin >> num2;
	std::cout << "\n";
	}

void get_only_input(){
	std::cout << "Enter number: 	";
	std::cin >> only;
	std::cout << "\n\n";
	}		

void pwr_of(){
	std::cout << "Enter power:	";
	std::cin >> pwr;
	std::cout << "\n\n";
	}

void addition(){
	result = num1 + num2;
	std::cout << "Result:		" << result;
	std::cout << "\n\n";
	}

void subtraction(){
	result = num1 - num2;
	std::cout << "Result:		"  << result;
	std::cout << "\n\n";
	}

void multiplication(){
	result = num1 * num2;
	std::cout << "Result:		"  << result;
	std::cout << "\n\n";
	}	

void division(){
	result = num1 / num2;
	std::cout << "Result:		"  << result;
	std::cout << "\n\n";
	}
		
void powers(){
	result = pow(only, pwr); 
	std::cout << "Result:		"  << result;
	std::cout << "\n\n";
	}

void square_root(){
	result = sqrt(only);
	std::cout << "Result:		"  << result;
	std::cout << "\n\n";
	}

int main(){
	get_function();
	while(func == 1 || func == 2 || func == 3 || func == 4){
		get_both_inputs();
		if(func == 1){
			addition();
			break;
		}else if(func == 2){
			subtraction();
			break;
		}else if(func == 3){
			multiplication();
			break;
		}else if(func == 4){
			if(num2 == 0){
				std::cout << "Divide by 0 error\n";
				break;
			}else{
				division();
				break;
				}
		}
	}
	while(func == 5 || func == 6){
		get_only_input();
		if(func == 5){
			pwr_of();
			powers();
			break;
		}else if(func == 6){
			if (only < 0){
				std::cout << "Imaginary number error\n";
				break;
			}else{
				square_root();
				break;
				}
			}
	}
	return 0;
}
		
	
